dtests/src/test/java/io/snappydata/hydra/cluster/SnappyStartUpTest.java

```diff
@@ -392,4 +392,169 @@ protected static void opsDuringServerHA_clusterRestart(String vmDir, String clie
+      }
+      if (indexNames.isEmpty()) {
+        throw new TestException("List of indexes against tableNames " +
+            "required for diagnostic query execution is not specified");
+      }
+      for (int i = 0; i < tableNames.size(); i++) {
+        String tableName = (String) tableNames.elementAt(i);
+        query = "select count(*), dsid() from sys.members m --GEMFIREXD-PROPERTIES withSecondaries=false \n , " + tableName + "  where dsid() = m.id";
+        rs = conn.createStatement().executeQuery(query);
+        while (rs.next()) {
+          numRowsPrimary = rs.getLong(1);
+          Log.getLogWriter().info("Qyery : " + query + " executed successfully and found " +
+              numRowsPrimary + " rows in primary buckets " +
+              queryExecutionTime + "  cluster restart.");
+        }
+        query = "select count(*), dsid() from sys.members m --GEMFIREXD-PROPERTIES withSecondaries=true \n , " + tableName + "  where dsid() = m.id";
+        rs = conn.createStatement().executeQuery(query);
+        while (rs.next()) {
+          numRowsPrimarySecondary = rs.getLong(1);
+          Log.getLogWriter().info("Qyery : " + query + " executed successfully and and found " +
+              numRowsPrimarySecondary + " rows in primary and secondary buckets " +
+              queryExecutionTime + " cluster " +
+              "restart.");
+        }
+        numRowsSecondary = numRowsPrimarySecondary - numRowsPrimary;
+        SnappyBB.getBB().getSharedMap().put("numRowsPrimary" +
+            queryExecutionTime + "Restart", numRowsPrimary);
+        SnappyBB.getBB().getSharedMap().put("numRowsSecondary" + queryExecutionTime + "Restart", numRowsSecondary);
+
+        for (int j = 0; j < indexNames.size(); j++) {
+          String indexName = (String) indexNames.elementAt(j);
+          String tableNameString = tableName.substring(tableName.indexOf(".") + 1);
+          if (indexName.contains(tableNameString)) {
+            query = "select count(*), dsid() from sys.members m , " + tableName + " " +
+                "--GEMFIREXD-PROPERTIES index=" + indexName + "  \n where dsid() = m.id ";
+            rs = conn.createStatement().executeQuery(query);
+            while (rs.next()) {
+              numRowsIndex = rs.getLong(1);
+              Log.getLogWriter().info("Qyery : " + query + " executed successfully and and found " +
+                  numRowsIndex + " rows in index:  " + indexName + " on table: " + tableName +
+                  queryExecutionTime + " cluster restart.");
+            }
+            SnappyBB.getBB().getSharedMap().put("numRows_" + indexName + "_" +
+                queryExecutionTime + "Restart", numRowsIndex);
+          }
+        }
+      }
+      closeConnection(conn);
+    } catch (SQLException e) {
+      SQLHelper.printSQLException(e);
+      throw new TestException("Not able to release the connection " + TestHelper.getStackTrace(e));
```

Comment:
SQLException can be thrown while getting a connection. Log message should be changed to make it a generic one.